### PR TITLE
KYRA: Add support for Chinese EOB2

### PR DESCRIPTION
**Pull request by phcoder**

### CODE REVIEW

**Comment by athrxx** [comment]
Wow, that's a great addition :-) . At first glance, the code changes seem unlikely to cause regressions to other versions. But it is too much to be sure, so we'll have to do some testing...

File: engines/kyra/graphics/screen_eob.cpp [hunk 1]
```diff
@@ -284,7 +326,75 @@ void Screen_EoB::loadShapeSetBitmap(const char *file, int tempPage, int destPage
...
+		if (in.getBit()) {
+			assert(destPtr < destEnd);
+			*destPtr++ = in.getByte();
+			continue;
+		}
+
+		if (!in.getBit()) {
+			lzLen = in.getBit() << 1;
+			lzLen |= in.getBit();
+			lzOffset = 0x100 - in.getByte();
```
Comment (athrxx) [inline engines/kyra/graphics/screen_eob.cpp:346]: MSVC throws an error here due to getBit() returning a bool.

Comment (phcoder) [reply engines/kyra/graphics/screen_eob.cpp:346]: Fixed

File: engines/kyra/engine/chargen.cpp [hunk 2]
```diff
@@ -1235,10 +1307,16 @@ int CharacterGenerator::modifyStat(int index, int8 *stat1, int8 *stat2) {
 	printStats(_activeBox, 3);
 	_vm->removeInputTop();
 
+	Screen::FontId of = _screen->setFont(Screen::FID_8_FNT);
```
Comment (athrxx) [inline engines/kyra/engine/chargen.cpp:1310]: This is the wrong font for the for the PC-98 games.
I have checked which one of the presets we can use: It seems that `Screen::FontId of = _screen->setFont(_vm->_invFont3);` will be correct for all games (Chinese, other DOS/Amiga, FM-Towns, PC-98 and SegaCD).

Comment (phcoder) [reply engines/kyra/engine/chargen.cpp]: Moved this code to Chinese block. Is it ok with you

Comment (athrxx) [reply engines/kyra/engine/chargen.cpp]: Sure, that's fine.

File: graphics/fonts/dosfont.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef GRAPHICS_FONTS_DOSFONT_H
+#define GRAPHICS_FONTS_DOSFONT_H
+
+#include <graphics/font.h>
```
Comment (sev-) [inline graphics/fonts/dosfont.h:25]: I'll fix this after the merge

**Comment by sev-** [comment]
Thank you!

**Pull request merged**